### PR TITLE
fix(sensing): finish #611 NaN-panic audit — 7 sites missed by #613

### LINKED ISSUE #611: adaptive_classifier.rs:94 — partial_cmp(b).unwrap() panics if NaN values enter the classifier
# Bug: `partial_cmp(b).unwrap()` panics if NaN values enter the classifier

## File
`v2/crates/wifi-densepose-sensing-server/src/adaptive_classifier.rs:94`

## Description

```rust
sorted.sort_by(|a, b| a.partial_cmp(b).unwrap());
```

The `.unwrap()` on `partial_cmp` **panics** if the slice contains a single `NaN` value, because `f64::partial_cmp(&NaN)` returns `None`.

## Impact

- CSI data from real ESP32 hardware can produce NaN (silent computation errors, division by zero in DSP, empty buffers)
- **Production code** called on every frame in the `classify()` path
- A single NaN frame kills the entire sensing server process

## Fix

```rust
sorted.sort_by(|a, b| a.partial_cmp(b).unwrap_or(std::cmp::Ordering::Equal));
```

## Notes

The same file already handles NaN correctly elsewhere (lines 149-150, 155 use `.unwrap_or(Ordering::Equal)`), making this an oversight.

## Severity

**Critical**  runtime panic on real-world data, process crash.
@bannned-bit

### PR DESCRIPTION
**Pull request by therahul-yo**
## Summary

Follow-up to #613 (which closed #611). #613 fixed `adaptive_classifier.rs:94` and explicitly stated the audit was complete. That audit grepped for the literal pattern `partial_cmp(b).unwrap()` and missed **seven additional production sites** that use comparator variants — all share the same crash class as #611 (NaN in `f64` data → `partial_cmp` returns `None` → `unwrap()` panics the sensing server).

## Sites fixed

| File | Line | Function | Why it matters |
|------|------|----------|----------------|
| `v2/.../adaptive_classifier.rs` | 205 | `AdaptiveModel::classify()` argmax over softmax probs | **Same per-frame hot path as #611.** A NaN in `raw_features` survives normalise (`NaN / x = NaN`) → logits → softmax (`exp(NaN)/exp(NaN) = NaN`), and the #613 IQR fix does not gate this branch. |
| `v2/.../adaptive_classifier.rs` | 480, 500 | `train_from_recordings()` argmax | Training accuracy / per-class accuracy loops. |
| `v2/.../main.rs` | 2446, 2449 | `count_persons_mincut` variance source/sink select | The outer `.unwrap_or((0, &0))` only catches an empty iterator — it cannot rescue a panic raised inside the comparator. A single NaN in `variances[]` aborts the process. |
| `v2/.../csi.rs` | 602, 605 | same logic, duplicated in `csi.rs` | Same as above. |

## Fix

Drop-in: swap `.unwrap()` for `.unwrap_or(std::cmp::Ordering::Equal)` inside the comparator closure. This is the same one-line behavioural change as #613 and matches the pattern the same files already use elsewhere (e.g. `adaptive_classifier.rs:149-150, 155`, and now `:94` from #613).

No API change, no test changes needed — the existing fix(sensing) commit precedent in CHANGELOG describes this as a behavioural fix only.

## Re-audit of remaining `partial_cmp(...).unwrap()` sites

After this PR, the only remaining `partial_cmp(…).unwrap()` matches in `v2/` are inside `#[cfg(test)]` / `#[test]` blocks where inputs are controlled and panic-on-NaN is acceptable:

- `wifi-densepose-signal/src/spectrogram.rs:269` — `#[test] fn test_single_frequency_peak`
- `wifi-densepose-pointcloud/src/depth.rs:234` — backproject test
- `ruv-neural/ruv-neural-signal/src/connectivity.rs:477` — coherence test
- `wifi-densepose-sensing-server/src/vital_signs.rs:737` — `#[test] fn test_fft_peak`

(#613's PR body listed the first three explicitly as benign; this PR confirms `vital_signs.rs:737` is the same category.)

## Test plan

- [ ] `cd v2 && cargo check -p wifi-densepose-sensing-server --no-default-features` (no Rust toolchain available in my environment — relying on CI).
- [ ] `cd v2 && cargo test --workspace --no-default-features` — 1,031+ passed expected, behavioural change should not affect any test that doesn't already inject NaN.
- [ ] Spot-grep `partial_cmp(...).unwrap()` after merge — only test/bench sites should remain.

Closes the audit gap from #611 / #613. No related open issue — filing this directly since the gap was caught by following the same grep methodology the original audit described.

### CODE REVIEW

**Review by ruvnet — APPROVED** [review]
Excellent audit — exactly the follow-up I should have done in #613. Specifically:

- Caught the **tuple-field comparator** form `partial_cmp(b.1).unwrap()` that my literal-string grep missed (`adaptive_classifier.rs:205` is on the same per-frame hot path as #611 — argmax over softmax — so any NaN from raw_feature → normalize → logits → softmax still crashed even after the #613 IQR fix)
- Caught the **field-projection** form `partial_cmp(&variances[b]).unwrap()` (`csi.rs:602/605`, `main.rs:2446/2449`) where the outer `unwrap_or((0, &0))` only catches an empty iterator, not a comparator panic. Same crash class disguised by a misleading rescue
- Correctly scoped: the remaining `partial_cmp(...).unwrap()` sites (`spectrogram.rs:269`, `depth.rs:234`, `connectivity.rs:477`, `vital_signs.rs:737`) are all inside `#[cfg(test)]` where inputs are controlled — left alone, as they should be
- All fixes use `unwrap_or(std::cmp::Ordering::Equal)` matching the pattern from #611/#613 and the in-file precedent

CHANGELOG entry is clear about what was missed and why. Merging.
